src/webui.py

```diff
@@ -154,6 +154,60 @@ def download_and_extract_zip(url, dir_name, progress=gr.Progress()):
     except Exception as e:
         raise gr.Error(str(e))
 
+def load_available_models(url):
+    with urllib.request.urlopen(url) as response:
+        text = response.read()
+
+    available_models_json = json.loads(text)
+    models_now = get_models_list(rvc_models_dir)
+    table_of_models = []
+   
+    
+    for model in available_models_json['voice_models']:
+        if not model['name'] in models_now:
+            model = [[model['name'],model['description'],model['credit'],model['url']]]
+            table_of_models+=model  
+    
+    all_tags = list(available_models_json['tags'].keys())
+    return gr.DataFrame.update(value=table_of_models),gr.CheckboxGroup.update(choices=all_tags)
+
+
+
+def search_models(url,tags=[],search_text=''):
+    with urllib.request.urlopen(url) as response:
+        text = response.read()
+
+    models_json = json.loads(text)
+    models_now = get_models_list(rvc_models_dir)
+    table_of_models = []
+    
+    for model in models_json['voice_models']:
+        if not model['name'] in models_now:
+            # check if only tags selected
+            if len(tags) > 0:
```

Comment:
This wouldn't work. It will also catch case where tags selected and search input is also present. This also means that it wouldn't reach line 200 at all.

Reply:
sheeesh! well, I've already said it, but I'm a bit of a noob, I don't have a particularly developed brain in this regard)
I'm a regular genshin player
If u can, just fix my code shit